aten/src/ATen/native/sparse/cuda/SparseBlasImpl.cpp

```diff
@@ -795,6 +795,56 @@ void addmm_out_sparse_csr(
   }
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~linalg_solve~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+template <typename scalar_t>
+void _apply_sparse_csr_lu_solve(
+  const Tensor& input,
+  const Tensor& other,
+  Tensor& result,
+  int& _singularity) {
+  auto values = input.values();
+  const scalar_t *values_data_ptr = values.data_ptr<scalar_t>();
+  auto crow_indices = input.crow_indices().to(kInt);
+  const int *crow_indices_data_ptr = crow_indices.data_ptr<int>();
+  auto col_indices = input.col_indices().to(kInt);
+  const int *col_indices_data_ptr = col_indices.data_ptr<int>();
+  cusolverSpHandle_t _handle;
```

Comment:
The handle was created but never destroyed.
Check out CuSparseHandlePool.cpp, CublasHandlePool.cpp, CusolverDnHandlePool.cpp how to properly handle the `cusolverSpHandle_t`. And a new file `CusolverSpHandlePool.cpp` needs to be created.

Reply:
Done. Thank you for the pointers on the files.